### ISSUE TITLE
Invalid `Git commit` returned by `notation version` installed via Homebrew

### ISSUE THREAD

**Issue by shizhMSFT** [notaryproject/notation#778]
### What is not working as expected?

The `Git commit` is not showing the tagged commit by running `notation version` when the `notation` binary is obtained from Homebrew using `brew install notation`.
```console
% notation version
Notation - a tool to sign and verify artifacts.

Version:     v1.0.0+Homebrew
Go version:  go1.21.0
Git commit:  Homebrew
```

### What did you expect to happen?

```console
% notation version
Notation - a tool to sign and verify artifacts.

Version:     v1.0.0+Homebrew
Go version:  go1.21.0
Git commit:  80e3fc4e2eeb43ac00bc888cf41101f5c56f1535
```
or
```console
% notation version
Notation - a tool to sign and verify artifacts.

Version:     v1.0.0+Homebrew
Go version:  go1.21.0
```

### How can we reproduce it?

```zsh
brew install notation
notation version
```

### Describe your environment

brew on macOS Ventura

### What is the version of your Notation CLI or Notation Library?

v1.0.0+Homebrew

**Comment by mintbomb27** [comment]
Do you require the full commit hash or just the short head will be enough?

**Comment by yizha1** [comment]
@mintbomb27  Is it OK to assign this issue to you for 1.1.0 installation, since you worked on the 1.0.0 installation?

**Comment by shizhMSFT** [comment]
> Do you require the full commit hash or just the short head will be enough?

As shown in the description, it should be the full commit hash.

**Comment by mintbomb27** [comment]
> @mintbomb27 Is it OK to assign this issue to you for 1.1.0 installation, since you worked on the 1.0.0 installation?

Sure! I'll work on this.

**Comment by mintbomb27** [comment]
Please review https://github.com/Homebrew/homebrew-core/pull/150339 @yizha1 @shizhMSFT

**Comment by chenrui333** [comment]
I am not sure why we care about the commitHash when you install via brew.

**Comment by shizhMSFT** [comment]
@mintbomb27 @chenrui333 The commit hash is a piece of duplicated information assuming the tag is never re-tagged. It is there for convenience for debugging, especially when opening a bug issue. GitHub will automatically add a hyperlink to the hash in the issue description like below

> Notation - a tool to sign and verify artifacts.
> 
> Version:     v1.0.0+Homebrew
> Go version:  go1.21.0
> Git commit:  80e3fc4e2eeb43ac00bc888cf41101f5c56f1535

If the commit hash is considered too duplicated, we can just remove that field by setting it to an empty string
```
-X #{project}/internal/version.GitCommit=
```
It is still better than having the string `Homebrew` in the "Git commit" field.

**Comment by chenrui333** [comment]
okay, let's just remove the `brew` string then, I still dont feel the strong need of having both release tag and commitHash (when they are pointing to the same source tree)

**Comment by mintbomb27** [comment]
This was fixed with [Homebrew/homebrew-core#150339](https://github.com/Homebrew/homebrew-core/pll/150339)
